apps/src/assetManagement/ImagePicker.jsx

```diff
@@ -0,0 +1,71 @@
+var AssetManager = require('./AssetManager.jsx');
+var IconLibrary = require('./IconLibrary.jsx');
+
+/**
+ * A component for managing hosted assets.
+ */
+module.exports = React.createClass({
+  propTypes: {
+    assetChosen: React.PropTypes.func,
+    typeFilter: React.PropTypes.string,
+    channelId: React.PropTypes.string.isRequired,
+    uploadsEnabled: React.PropTypes.bool.isRequired
+  },
+
+  getInitialState: function () {
+    return {mode: 'files'};
+  },
+
+  setIconMode: function () {
+    this.setState({mode: 'icons'});
+  },
+
+  setFileMode: function () {
+    this.setState({mode: 'files'});
+  },
+
+  render: function () {
+
+    var modeSwitch, title = this.props.assetChosen ?
+      <p className="dialog-title">Choose Assets</p> :
+      <p className="dialog-title">Manage Assets</p>;
+
+    if (this.props.assetChosen) {
+      var fileMode = this.state.mode === 'files';
```

Comment:
I might call this something like `isFileMode` to make it clear that I expect this to be a boolean, rather than a mode.